addons/sale_timesheet/models/project.py

```diff
@@ -175,6 +176,43 @@ def _compute_sale_line_id(self):
             ], limit=1)
             project.sale_line_id = sol or project.sale_line_employee_ids.sale_line_id[:1]  # get the first SOL containing in the employee mappings if no sol found in the search
 
+    @api.depends('sale_line_id.product_uom_qty', 'sale_line_id.product_uom')
+    def _compute_allocated_hours(self):
+        sale_orders = self.sale_order_id | self.tasks.sale_order_id
+        sols_read_group = self.env['sale.order.line'].read_group([
+            ('order_id', 'in', sale_orders.ids), ('is_service', '=', True),
+            ('is_downpayment', '=', False), ('product_id.project_template_id', '=', False),
```

Comment:
Do you know why, functionally speaking, did @aju-odoo asked you to add this in the domain?
`('product_id.project_template_id', '=', False),`

Reply:
@IT-Ideas , Yes here is the point: `https://watch.screencastify.com/v/VMbK7srKqBBdVYaxfnXN project.project > allocated hours: don't take into account services that have a project template set in the calculation`

Thanks

Reply:
```suggestion
            ('is_downpayment', '=', False),
```

Reply:
@IT-Ideas , as discussed with @aju-odoo ,
If the user has explicitly set allocated_hours on the template, I believe we should copy that template's allocated_hours. If the user has left the allocated_hours of the template to 0, then we should calculate it based on the qty of the sale order line.

so we will update the domain accordingly.  is it ok ?

Thanks